lib/Db/TagMapper.php

```diff
@@ -248,4 +225,26 @@ public function createDefaultTags(MailAccount $account): void {
 			$this->insert($entity);
 		}
 	}
+
+	public function deleteOrphans() {
+		$qb = $this->db->getQueryBuilder();
+		$subQb1 = $this->db->getQueryBuilder();
+		$dqb = $this->db->getQueryBuilder();
+
+		// get lowest ids
+		$groupByQuery = $subQb1->select($qb->func()->min('id'))
+			->from('mail_message_tags')
+			->groupBy('imap_message_id','tag_id')
+			->having('COUNT(id)>1');
+
+		// select those for deleting
+		$select = $qb->select('*')
+			->from($qb->createFunction('(' . $groupByQuery->getSql() .')'), 'temp');
```

Comment:
is a select from a sub query supported by all of our supported databases?

Reply:
on a second thought, does that query do anything useful? you select * from another query. can't you just use that other query?